src/cpp/server/server_cc.cc

```diff
@@ -1364,20 +1367,39 @@ grpc::ServerInitializer* Server::initializer() {
   return server_initializer_.get();
 }
 
-grpc::CompletionQueue* Server::CallbackCQ() {
-  // TODO(vjpai): Consider using a single global CQ for the default CQ
-  // if there is no explicit per-server CQ registered
-  grpc::internal::MutexLock l(&mu_);
-  if (callback_cq_ == nullptr) {
+static constexpr unsigned kDefaultCallbackCqShards = 32;
+void GPR_ATTRIBUTE_NOINLINE Server::MakeCallbackCQShardsLocked() {
+  const unsigned num_cores = gpr_cpu_num_cores();
+  const size_t num_cb_cq_shards =
+      num_cores == 0 ? kDefaultCallbackCqShards
+                     : static_cast<size_t>(pow(
```

Comment:
Can we avoid using floating-point math for round-off reasons?

Reply:
Sure, will do - easy enough since I just want log_2.